### PR TITLE
#138 table of contents (All Java chapters)

### PR DESCRIPTION
**Pull request by iskandarzulkarnaien**
Adds table of contents (toc) to articles without one and hides them except for mobile and print view as discussed in #138 . 

This PR affects the below chapters under `Programming Languages`. Kindly use the below checklist for reviewing.

Items to test for each chapter:
- [ ] Web view (Expected: in-article toc absent)
- [ ] Mobile view (Expected: in-article toc present and **links clickable**, right nav panel absent)
- [ ] Print view (Expected: in-article toc present, left and right nav panels absent)

Chapters:
- [ ] Java Concurrency
- [ ] Java Streams
- [ ] Java Synchronization
- [ ] Java Reflection
- [ ] JUnit

Devtools shortcuts
Firefox:
<kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>M</kbd>

Chrome:
<kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>I</kbd>

### CODE REVIEW

**Comment by iskandarzulkarnaien** [comment]
@xantho09 , @geshuming could both of you review my PR if you have the time. Thanks!

**Comment by iskandarzulkarnaien** [comment]
@jamessspanggg , @madanalogy if you guys could spare the time I'd appreciate a review. Thanks!

File: contents/java/JUnit.md [hunk 1]
```diff
@@ -17,6 +17,17 @@
 
 Reviewers: [Aadyaa Maddi](https://github.com/amad-person), [Marvin Chin](https://github.com/marvinchin)
 
+<box id="article-toc">
+
+* [What is JUnit?‎](#what-is-junit)
+* [Why JUnit?‎](#why-junit)
+* [JUnit Features‎](#junit-features)
+  * [Assertions API‎](#assertions-api)
+  * [Before/After hooks‎](#before-after-hooks)
```
Comment (madanalogy) [inline contents/java/JUnit.md]:
```suggestion
  * [Before/After Hooks‎](#before-after-hooks)
```

File: contents/java/JavaReflections.md [hunk 2]
```diff
@@ -15,6 +15,17 @@
...
 
+<box id="article-toc">
+
+* [Reflection‎](#reflection)
+  * [What is Java reflections‎](#what-is-java-reflections)
+  * [The Basics of Reflections‎](#the-basics-of-reflections)
+  * [Getting Started‎](#getting-started)
+  * [Applications‎](#applications)
+  * [Disadvantages of reflections‎](#disadvantages-of-reflections)
+  * [Further Resources for reflections‎](#further-resources-for-reflections)
```
Comment (madanalogy) [inline contents/java/JavaReflections.md]:
```suggestion
  * [Disadvantages of Reflections‎](#disadvantages-of-reflections)
  * [Further Resources for Reflections‎](#further-resources-for-reflections)
```

File: contents/java/JavaSynchronization.md [hunk 3]
```diff
@@ -15,6 +15,21 @@
...
+* [Basic Syntax‎](#basic-syntax)
+	* [synchronized statement‎](#synchronized-statement)
+	* [synchronized method‎](#synchronized-method)
+* [More features of Java synchronization‎](#more-features-of-java-synchronization)
+	* [wait() and notifyAll()‎](#wait-and-notifyall)
+		* [Understand which object is used as Monitor‎](#understand-which-object-is-used-as-monitor)
+		* [Check condition for wait() with while loop‎](#check-condition-for-wait-with-while-loop)
+		* [Breakpoint of notifyAll()‎](#breakpoint-of-notifyall)
+	* [notify()‎](#notify)
+* [Related areas‎](#related-areas)
```
Comment (madanalogy) [inline contents/java/JavaSynchronization.md]:
```suggestion
	* [Synchronized Statement‎](#synchronized-statement)
	* [Synchronized Method‎](#synchronized-method)
* [More Features of Java Synchronization‎](#more-features-of-java-synchronization)
	* [wait() and notifyAll()‎](#wait-and-notifyall)
		* [Understand Which Object is Used as Monitor‎](#understand-which-object-is-used-as-monitor)
		* [Check Condition for wait() with While Loop‎](#check-condition-for-wait-with-while-loop)
		* [Breakpoint of notifyAll()‎](#breakpoint-of-notifyall)
	* [notify()‎](#notify)
* [Related Areas‎](#related-areas)
```

File: contents/java/streams-an-introduction.md [hunk 4]
```diff
@@ -15,35 +15,34 @@
...
+    * [Collect](#collect)
+  * [Drawbacks and Pitfalls](#drawbacks-and-pitfalls)
+    * [Long, complicated lambda expressions](#long-complicated-lambda-expressions)
+    * [Difficulty in optimising stream performance](#difficulty-in-optimising-stream-performance)
+* [Resources](#resources)
+  * [Functional Interfaces](#functional-interfaces)
+  * [Method References](#method-references)
+  * [Lambda Expressions](#lambda-expressions)
+  * [Stream](#stream)
+    * [Common pitfalls](#common-pitfalls)
```
Comment (madanalogy) [inline contents/java/streams-an-introduction.md]:
```suggestion
    * [Common Pitfalls](#common-pitfalls)
```

**Review by madanalogy — CHANGES_REQUESTED** [review]
Figured it might be worthwhile to also follow titleCase as per https://github.com/se-edu/learningresources/issues/164

**Review by jamessspanggg — CHANGES_REQUESTED** [review]
Just one nit apart from the comments given above:

File: contents/java/JavaReflections.md [hunk 5]
```diff
@@ -15,6 +15,17 @@
 
 Authors: [Jeremy Goh](https://github.com/MightyCupcakes), [Yong Zhi Yuan](https://github.com/Zhiyuan-Amos)
 
+<box id="article-toc">
+
+* [Reflection‎](#reflection)
+  * [What is Java reflections‎](#what-is-java-reflections)
+  * [The Basics of Reflections‎](#the-basics-of-reflections)
+  * [Getting Started‎](#getting-started)
+  * [Applications‎](#applications)
```
Comment (jamessspanggg) [inline contents/java/JavaReflections.md:24]: can add the subsection of applications to the side TOC and main TOC?

Comment (iskandarzulkarnaien) [reply contents/java/JavaReflections.md:24]: Done!

**Comment by iskandarzulkarnaien** [comment]
@madanalogy , @jamessspanggg could I get a re-review? Thanks!

Also @madanalogy I'll look through all the content pages again after all the other related PRs have been merged, just to be sure the TitleCasing is consistently applied

**Review by madanalogy — APPROVED** [review]

File: contents/java/JavaSynchronization.md [hunk 6]
```diff
@@ -15,6 +15,21 @@
 
 Authors: [Boxin](https://github.com/boxin-yang)
 
+<box id="article-toc">
+
+* [Introduction‎](#introduction)
+* [Basic Syntax‎](#basic-syntax)
+	* [Synchronized Statement‎](#synchronized-statement)
+	* [Synchronized Method‎](#synchronized-method)
+* [More Features of Java Synchronization‎](#more-features-of-java-synchronization)
```
Comment (jamessspanggg) [inline contents/java/JavaSynchronization.md]: can u standardise the indentation?

**Review by jamessspanggg — COMMENTED** [review]
Just one nit

File: contents/java/JavaSynchronization.md (hunk 6 above)
Comment (iskandarzulkarnaien) [reply contents/java/JavaSynchronization.md]: changed to 2 spaces instead of using tabs. Thanks

**Comment by iskandarzulkarnaien** [comment]
@jamessspanggg have resolved the issue. Thanks!

**Review by jamessspanggg — APPROVED** [review]

**Comment by iskandarzulkarnaien** [comment]
Thanks @madanalogy and @jamessspanggg !

@damithc this PR is ready to be merged

**Pull request merged**